### ISSUE TITLE
GitHub SSH failure due to `insteadOf` git configuration

### ISSUE THREAD

**Issue by nyurik** [cargo-generate/cargo-generate#874]
**Describe the bug**
If a user has automatic git redirect from https to ssh, `cargo generate` fails with the `invalid or unknown remote ssh hostkey` error.

**make sure you**
- [x] I'm on the latest rust version `rustup update`
- [x] I'm on the latest cargo-generate version `cargo install cargo-generate`
- [x] In case you have an own `.gitconfig`, provide all sections that contains one of: `insteadOf`, `url`

**To Reproduce**
* In `~/.gitconfig` (on Linux), add this section:

```
[url "ssh://git@github.com/"]
	insteadOf = https://github.com/
```
* Make sure your ssh key is properly configured in your github account, and it matches your user name
* First test that it works as is, and observe that the remote is using SSH rather than https despite cloning http:

```
❯ git clone https://github.com/Rahix/avr-hal-template
Cloning into 'avr-hal-template'...
...
❯ cd avr-hal-template/
❯ git remote -v
origin	ssh://git@github.com/Rahix/avr-hal-template (fetch)          <--------------  THESE TWO MUST BE ssh://git@...
origin	ssh://git@github.com/Rahix/avr-hal-template (push)
❯ cd ..
❯ rm -rf avr-hal-template/
```

* Run `cargo generate --git https://github.com/rahix/avr-hal-template`

```
❯ cargo generate --git https://github.com/rahix/avr-hal-template
Error: Please check if the Git user / repository exists.

Caused by:
    invalid or unknown remote ssh hostkey; class=Ssh (23); code=Certificate (-17)
```

**Expected behavior**
generation should start

**Desktop (please complete the following information):**
 - OS: Linux Mint
 - Version: 21.1
 - Build tools: a lot of them
 - Shell: bash
 - `rustc --version`: rustc 1.67.1 (d5a82bbd2 2023-02-07)
 - `cargo generate --version`: cargo generate-generate 0.17.6

**Comment by Yiwen-Gao** [comment]
I have the same Git setting to use SSH instead of HTTPS and ran into the same issue with:
```
cargo generate --git https://github.com/CosmWasm/cw-template.git --name cw --verbose
```
but my error message was:
```
Error: Please check if the Git user / repository exists.

Caused by:
    all authentication attempts failed
```

As a workaround, I commented out the `insteadOf` directive in my `~/.gitconfig`, and `cargo generate` was able to authenticate and download everything correctly.

**Comment by c98tristan** [comment]
Any update?
I got this issue on Aug 2024 with this error
```
cargo generate --git https://github.com/CosmWasm/cw-template.git --branch 1.0 --name my-first-contract
Error: Please check if the Git user / repository exists.

Caused by:
    could not read from remote repository; class=Net (12); code=Eof (-20)
```

**Comment by ForsakenHarmony** [comment]
A workaround is to use `http://` instead of `https://`, but it would be nice if this got addressed at some point

**Comment by sassman** [comment]
BTW: the right way to configure the "instead URL" in your git config is:

```
[url "ssh://git@github.com:"]
    insteadOf = https://github.com/
```

Because the repo owner is behind a `:` instead of a `/`
e.g. 
<img width="388" alt="grafik" src="https://github.com/user-attachments/assets/d13f9815-0b1a-416e-880c-fcc5454ad9db">

**Comment by sassman** [comment]
This will be released with `v0.22.0`
